### DISCUSSION TITLE
mole fraction vary from  path

### DISCUSSION THREAD

**Discussion by windofport** [radis/radis#715] [Q&A]
how  to  use  radis when the mole fraction vary from path

**Comment by erwanp** [comment]
Do you mean when the mole fraction varies along the path line ?

Then, you should use multiple "Spectrum" to objects and add them in series, each of them with a defined column length.
See the Line of sight module : https://radis.readthedocs.io/en/latest/los/los.html#how-to-combine-slabs

```
slab_list=[s0, S1, S2, s3,...]
los = SerialSlabs(slab_list)
```

Important note : if only the mole fraction change (and neither the temperature nor the pressure) you can recompute one spectrum instantly, with a usually negligible error. This might be useful if you're computing hundreds of spectra. See https://radis.readthedocs.io/en/latest/spectrum/spectrum.html#rescale-spectrum-with-new-mole-fraction
